UNITTESTS/README.md

```diff
@@ -0,0 +1,65 @@
+## Stubs, mocks and fakes in Mbed OS
```

Comment:
Better we move this readme content into our official documentation https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/debugging-testing/testing/unit_testing.md page 

Note:
Already this PR for updating https://github.com/ARMmbed/mbed-os-5-docs/pull/1427 content based on new unittest CMake build system

Reply:
I already have a PR for documentation of ble unit testing: https://github.com/ARMmbed/mbed-os-5-docs/pull/1399/files

Reply:
I requested Readme to introduce stubs and fakes (I was thinking putting it into stubs/fakes folders). It can contain basic info and reference to our docs pages. Basically if I fetch the repo and start browsing sources, I can read readme and browse files and contribute.

@paul-szczepanek-arm Lets move this readme to your 1399 PR. It contains valuable information.

@rajkan01 Are you proposing to completely remove this readme (move everything to docs) ?

Reply:
OK, so you don't want the readme in mbed-os?

Reply:
@0xc0170 @paul-szczepanek-arm Yes, I have only removed the existing `Readme` from the `UNITTESTS` directory as part of `Refactor unittest cmake PR` as content gets duplicated and also official vs local readme content mismatch so better to move this readme content into the official Mbed OS documentation